### ISSUE TITLE
RuntimeError: Calculated padded input size per channel: (3 x 3). Kernel size: (4 x 4). Kernel size can't be greater than actual input size

### ISSUE THREAD

**Issue by jpnavarro-nv** [junyanz/pytorch-CycleGAN-and-pix2pix#776]
Hi.
I'm running the following training command:
`python train.py --model pix2pix --name phase_pix2pix --dataroot ./datasets/phase --batch_size 32 --gpu_ids 0,1,2,3 --preprocess scale_width_and_crop --load_size 512 --crop_size 128 --direction AtoB`

And getting the following error:
`RuntimeError: Calculated padded input size per channel: (3 x 3). Kernel size: (4 x 4). Kernel size can't be greater than actual input size`

Using the latest master revision, within this is the Dockerfile to generate the environment:

```
FROM nvidia/cuda:9.0-base

RUN apt update && apt install -y wget unzip curl bzip2 git
RUN curl -LO http://repo.continuum.io/miniconda/Miniconda3-latest-Linux-x86_64.sh
RUN bash Miniconda3-latest-Linux-x86_64.sh -p /miniconda -b
RUN rm Miniconda3-latest-Linux-x86_64.sh
ENV PATH=/miniconda/bin:${PATH}
RUN conda update -y conda

RUN conda install -y pytorch torchvision -c pytorch
RUN mkdir /workspace/ && cd /workspace/ && git clone https://github.com/junyanz/pytorch-CycleGAN-and-pix2pix.git && cd pytorch-CycleGAN-and-pix2pix && pip install -r requirements.txt

WORKDIR /workspace
```
Any clue?

**Comment by junyanz** [comment]
The default pix2pix generator only supports 256x256. However, your crop_size is 128. You can try three things (1)  increase your crop_size to 256, (2) remove 1-2 layers from the default pix2pix generator, (3) use a resnet generator such as resnet_9blocks or resnet_6blocks.

**Comment by jpnavarro-nv** [comment]
Hi @junyanz ! Changing the crop size to 256 solved the issue. Many thanks for your reply.

**Comment by zhangjy328** [comment]
@junyanz Hello, I encountered the same problem when I tested the pix2pix model. When I tested it normally, it was ok, when I ran down the code:

`python test.py --dataroot ./datasets/underwater_pix/test_AB --name underwater_pix2pix --model pix2pix --direction BtoA`

but this problem arose after I added the --preprocess none. As follows:

`python test.py --dataroot ./datasets/underwater_pix/test_AB --name underwater_pix2pix --model pix2pix --direction BtoA --preprocess none `

**When I tested with the image size of 640×480, an error was reported:**
  File "D:\models\enhance_models\pytorch-CycleGAN-and-pix2pix-master\models\networks.py", line 536, in forward
    return torch.cat([x, self.model(x)], 1)
RuntimeError: Sizes of tensors must match except in dimension 1. Expected size 3 but got size 2 for tensor number 1 in the list.


**When I tested with the image size of 256×256, an error was reported:**
  File "D:\Anaconda3\envs\deeplab\lib\site-packages\torch\nn\modules\conv.py", line 459, in _conv_forward
    return F.conv2d(input, weight, bias, self.stride,
RuntimeError: Calculated padded input size per channel: (4 x 3). Kernel size: (4 x 4). Kernel size can't be greater than actual input size
